subsys/bluetooth/host/conn.c

```diff
@@ -683,6 +659,25 @@ static int send_frag(struct bt_conn *conn,
 		net_buf_add_mem(frag, buf->data, frag_len);
 		net_buf_pull(buf, frag_len);
 	} else {
+		if (tx_data(buf)->cb) {
+			tx = conn_tx_alloc();
+			if (!tx) {
+				LOG_DBG("Unable to allocate TX context");
+				return -ENOBUFS;
```

Comment:
runtime assert (should never fail or block).
static assert to have same n of ctlr bufs as tx contexts. (noop `CONFIG_BT_CONN_TX_MAX`).